FilePersistence/src/version_control/Commit.cpp

```diff
@@ -103,4 +103,88 @@ bool Commit::getFileContent(QString fileName, const char*& content, int& content
 		return false;
 }
 
+QStringList Commit::regexSearchNodeData(QString id) const
+{
+	QStringList matches;
+	for (auto file : files())
+	{
+		int index = 0;
```

Comment:
Don't call things using generic names like `index` or `flag`. What is this index? What does the flag actually indicate? Please use descriptive names.